servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -128,8 +131,9 @@ final class DefaultLoadBalancer<ResolvedAddress, C extends LoadBalancedConnectio
             final HostSelector<ResolvedAddress, C> hostSelector,
             final ConnectionFactory<ResolvedAddress, ? extends C> connectionFactory,
             final int linearSearchSpace,
+            final LoadBalancerObserver<ResolvedAddress> loadBalancerObserver,
             @Nullable final HealthCheckConfig healthCheckConfig,
-            @Nullable final LoadBalancerObserver<ResolvedAddress> loadBalancerObserver) {
+            @Nullable final Supplier<HealthChecker> healthCheckerFactory) {
```

Comment:
`Supplier<HealthChecker> -> Supplier<HealthChecker<ResolvedAddress>>`, we should also add parameters for `LoadBalancingPolicy`